asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py

```diff
@@ -0,0 +1,778 @@
+import argparse
+import logging
+import shutil
+from datetime import datetime
+from functools import partial
+from pathlib import Path  # noqa: F401
+from typing import List  # noqa: F401
+
+import dask
+import yaml
+from asapdiscovery.data.execution_utils import get_interfaces_with_dual_ip
+from asapdiscovery.data.logging import FileLogger
+from asapdiscovery.data.openeye import (
+    oechem,
+    save_openeye_pdb,
+    save_openeye_sdf,
+    split_openeye_design_unit,
+)
+from asapdiscovery.data.schema import CrystalCompoundData, ExperimentalCompoundData
+from asapdiscovery.data.utils import (
+    exp_data_to_oe_mols,
+    is_valid_smiles,
+    oe_load_exp_from_file,
+)
+from asapdiscovery.dataviz.gif_vis import GIFVisualiser
+from asapdiscovery.dataviz.html_vis import HTMLVisualiser
+from asapdiscovery.docking import make_docking_result_dataframe
+from asapdiscovery.docking import prep_mp as oe_prep_function
+from asapdiscovery.docking.mcs import rank_structures_openeye  # noqa: F401
+from asapdiscovery.docking.mcs import rank_structures_rdkit  # noqa: F401
+from asapdiscovery.docking.scripts.run_docking_oe import mp_func as oe_docking_function
```

Comment:
Let's move `mp_func` out of the `run_docking_oe` script, and instead put it somewhere in the library. You can then import it into both `run_docking_oe` and this script.